text/0000-match-ergonomics.md

```diff
@@ -3,11 +3,15 @@
 - RFC PR: (leave this empty)
 - Rust Issue: (leave this empty)
 
+I'm writing comments inline (not making changes), hopefully they'll stand out in the diff.
+
 # Summary
 [summary]: #summary
 
 Better ergonomics for pattern-matching on references.
 
+AIUI, this will only apply to match and similar expressions, not pattern matching in general?
```

Comment:
1. > How does this look when calling a function?

I wouldn't expect `fn` call sites to be affected. Declaring `fn` args would be affected, though:
```rust
struct Bar<T>(T);
fn foo(Bar(a): &Bar<i32>) { ... }
// desugars to
fn foo(&Bar(ref a): &Bar<i32>) { ... }
```

2. > Does it introduce auto-ref in places it didn't exist before? Is it still backwards compatible?

The compiler will only introduce autoderef and binding modes when it encounters code that matches a reference with a non-reference pattern, i.e. code that wouldn't have compiled before. I don't think the type of the pattern (i.e. match, if let, let, etc.) matters.

3. > How will this work with multiple match arms with multiple patterns and/or guards?

Each match is treated independently, as some of the examples show. i.e. each branch uses its own default binding mode starting at `move`, which adjusts as patterns match inwards.